### ISSUE TITLE
Improve "jsx-no-lambda" rule detection

### ISSUE THREAD

**Issue by no23reason** [palantir/tslint-react#29]
This rule is reporting false positives in situations where the lambda is used as a parameter to a function returning some value. For example:

``` tsx
interface Item { value: number; }

<Currency amount={ _.sumBy(items, (item: Item) => item.value) } />
```

This usage does not have any negative implications with regards to the `shallowCompare` render optimisation as the lambda is never actually compared.

I am aware that simple workaround would often be to compute the value into a variable and then use it like this:

``` tsx
interface Item { value: number; }

const total = _.sumBy(items, (item: Item) => item.value);
<Currency amount={ total } />
```

However, that seems tedious and unnecessary.

**Comment by JKillian** [comment]
👍  Seems like a reasonable enhancement @no23reason
